src/CommunityToolkit.Datasync.Client/Offline/Operations/ExecutableOperation.cs

```diff
@@ -22,19 +22,22 @@ internal abstract class ExecutableOperation
     /// </summary>
     /// <param name="baseAddress">The base address from the client.</param>
     /// <param name="relativeOrAbsoluteUri">A relative or absolute URI</param>
+    /// <param name="itemId">The id of the item to append to the URI.</param>
     /// <returns></returns>
-    internal static Uri MakeAbsoluteUri(Uri? baseAddress, Uri relativeOrAbsoluteUri)
+    internal static Uri MakeAbsoluteUri(Uri? baseAddress, Uri relativeOrAbsoluteUri, string? itemId = null)
     {
+        itemId = string.IsNullOrWhiteSpace(itemId) ? string.Empty : $"/{itemId}";
```

Comment:
@adrianhall the separator is set here if the itemId is specified.

Reply:
Ah - I get it.  Although, use of IsNullOrWhiteSpace does change the semantics of the method.  If someone submits " ", I believe it gets rejected before this point, so should be ok.

Assuming the tests pass, I'll merge this.